Source/Pawnmorphs/Esoteria/Hediff_AddedMutation.cs

```diff
@@ -148,17 +156,7 @@ public override string DebugString()
         ///     Gets a value indicating whether should be removed.
         /// </summary>
         /// <value><c>true</c> if should be removed; otherwise, <c>false</c>.</value>
-        public override bool ShouldRemove
-        {
-            get
-            {
-                foreach (HediffComp hediffComp in comps.MakeSafe())
-                    if (hediffComp.CompShouldRemove)
-                        return true;
-
-                return shouldRemove;
-            }
-        }
+        public override bool ShouldRemove => shouldRemove;

```

Comment:
Cached ShouldRemove to avoid checking for comps every tick for every mutation on every pawn.